integrations/tensorflow/e2e/math/math_test.py

```diff
@@ -0,0 +1,561 @@
+    }
+
+flags.DEFINE_list(
+    "functions", "abs",
+    f"Any of {list(FUNCTION_TO_CONFIGS.keys())}. If more than one function is "
+    "provided then len(--target_backends) must be one.")
+flags.DEFINE_bool(
+    "dynamic_dims", False,
+    "Whether or not to compile the layer with dynamic dimensions.")
+flags.DEFINE_bool(
+    "test_complex", False,
+    "Whether or not to test or ignore function signatures with complex types.")
+flags.DEFINE_bool(
+    'list_functions_with_complex_tests', False,
+    'Whether or not to print out all functions with complex inputs '
+    '(and skip running the tests).')
+
+
+def _complex_wrapper(function):
+  """Wraps a tf.function to allow compiling functions of complex numbers."""
+
+  def decorator(*args, **kwargs):
+    inputs = []
+    for real, imag in zip(args[::2], args[1::2]):
+      inputs.append(tf.complex(real, imag))
+    result = function(*inputs, **kwargs)
+    # Directly returning a complex number causes an error.
+    return tf.math.real(result) + tf.math.imag(result)
+
+  return decorator
+
+
+def convert_if_complex(function, config: Config):
+  """Compatibility layer for testing complex numbers."""
+  if is_complex(config):
+    if not all([spec.dtype.is_complex for spec in config.signature]):
+      raise NotImplementedError("Signatures with mixed complex and non-complex "
+                                "tensor specs are not supported.")
+
+    # Rewrite the signature, replacing all complex tensors with pairs of real
+    # and imaginary tensors.
+    signature = []
+    for spec in config.signature:
+      new_dtype = tf.float32 if spec.dtype.size == 8 else tf.float64
+      signature.append(tf.TensorSpec(spec.shape, new_dtype))
+      signature.append(tf.TensorSpec(spec.shape, new_dtype))
+
+    return _complex_wrapper(function), signature
+  else:
+    return function, config.signature
+
+
+def _make_dims_dynamic(spec: tf.TensorSpec) -> tf.TensorSpec:
+  return tf.TensorSpec([None] * len(spec.shape), spec.dtype)
+
+
+def create_function_unittest(config: Config, function_name: str,
+                             exported_name: str) -> tf.function:
+  """Creates a tf_function_unittest from the provided Config."""
+  tf_math_function = getattr(tf.math, function_name)
+  tf_math_function, signature = convert_if_complex(tf_math_function, config)
+  wrapped_function = lambda *args: tf_math_function(*args, **config.kwargs)
+
+  if FLAGS.dynamic_dims:
+    signature = tf_utils.apply_function(signature, _make_dims_dynamic)
+
+  return tf_test_utils.tf_function_unittest(
+      input_signature=signature,
+      input_generator=config.generator,
+      input_args=config.args,
+      name=exported_name)(wrapped_function)
+
+
+def create_unittests(module: Union["TfMathModule", Type["TfMathModule"]]):
+  """Creates all tf_function_unittests to test and puts them on module."""
+  for function_name in FLAGS.functions:
+    # pytype: disable=attribute-error :\
```

Comment:
I assume this is because you define this with some entries having no-dict values and only later. What about instead doing this such that `FUNCTION_TO_CONFIGS` has a fixed type. A couple of options:

1. Use two variables
2. define a function that does the normalization and normalize the anonymous dict in the assignment of `FUNCTION_TO_CONFIGS`

Reply:
Did the latter!